terraform/aks/locals.tf

```diff
@@ -10,4 +10,5 @@ locals {
     REDIS_WORKER_URL = module.redis_worker.url
   }
   app_env_values = yamldecode(file(var.app_config_file))[var.app_environment]
+  infra_secrets  = yamldecode(data.azurerm_key_vault_secret.infra_secrets.value)
```

Comment:
You could probably use the secrets module to simplify this avoid needing the `data` blocks, something like:

```tf 
module "secrets" {
  source = "git::https://github.com/DFE-Digital/terraform-modules.git//aks/secrets?ref=stable"

  azure_resource_prefix = var.azure_resource_prefix
  service_short         = var.service_short
  config_short          = var.config_short
}

infra_secrets  = yamldecode(module.secrets.map[var.key_vault_secret_infra_name)
```

Reply:
Updated as suggested.